pennylane/io/qualtran_io.py

```diff
@@ -0,0 +1,203 @@
+            temp_registers = _get_named_registers(cbloq.signature.lefts())
+            soq_to_wires = {
+                Soquet(LeftDangle, idx=idx, reg=reg): list(temp_registers[reg.name])
+                for reg in cbloq.signature.lefts()
+                for idx in reg.all_idxs()
+            }
+
+            for binst, pred_cxns, succ_cxns in cbloq.iter_bloqnections():
+                in_quregs = {
+                    reg.name: np.empty((*reg.shape, reg.bitsize), dtype=object).flatten()
+                    for reg in binst.bloq.signature.lefts()
+                }
+                # The out_quregs inform us of the total # of wires in the circuit to account for
+                # wires that are split or allocated in the cbloq
+                out_quregs = {
+                    reg.name: np.empty((*reg.shape, reg.bitsize), dtype=object).flatten()
+                    for reg in binst.bloq.signature.rights()
+                }
+
+                # This is to track the number of wires defined at the LeftDangle stage
+                # so if we need to add more wires, we know what index to start at
+                soq_to_wires_len = 0
+                if len(soq_to_wires.values()) > 0:
+                    try:
+                        soq_to_wires_len = list(soq_to_wires.values())[-1][-1] + 1
+                    except (IndexError, TypeError):
+                        soq_to_wires_len = list(soq_to_wires.values())[-1] + 1
+
+                for pred in pred_cxns:
+                    soq = pred.right
+                    soq_to_wires[soq] = soq_to_wires[pred.left]
+                    in_quregs[soq.reg.name][soq.idx] = soq_to_wires[soq]
+
+                for succ in succ_cxns:
+                    soq = succ.left
+                    if soq.reg.side == Side.RIGHT:
+                        # When in_quregs != out_quregs, it means that there are wires unaccounted
+                        # for. We account for these wires and update soq_to_wires and in_quregs
+                        # accordingly.
+                        if len(in_quregs) != len(out_quregs):
+                            total_elements = np.prod(soq.reg.shape) * soq.reg.bitsize
+                            ascending_vals = np.arange(
+                                soq_to_wires_len,
+                                total_elements + soq_to_wires_len,
+                                dtype=object,
+                            )
+                            in_quregs[soq.reg.name] = ascending_vals.reshape(
+                                (*soq.reg.shape, soq.reg.bitsize)
+                            )
+                        soq_to_wires[soq] = in_quregs[soq.reg.name][soq.idx]
+
+                total_wires = [w for ws in in_quregs.values() for w in list(ws.flatten())]
+
+                mapped_wires = [wires[idx] for idx in total_wires]
+                op = binst.bloq.as_pl_op(mapped_wires)
```

Comment:
Might want to add URL for this as a comment.

Reply:
URL to qualtran?